tg_bot/modules/locks.py

```diff
@@ -208,6 +283,17 @@ def del_lockables(bot: Bot, update: Update):
                         chat.kick_member(new_mem.id)
                         message.reply_text("Only admins are allowed to add bots to this chat! Get outta here.")
             else:
+                #allow whitelisted URLs
+                if lockable == 'url':
+                    entities = set(message.parse_entities([MessageEntity.URL]).values())
+
+                    # MessageEntity.TEXT_LINK could be added in the filter above, but would return the text, not the
+                    # url. So I must add all entities that have a 'url' field separately
+                    entities = entities | set(entity.url for entity in message.entities if entity.url)
```

Comment:
Would it be possible to reduce this to a single call to set()? are you sure the contents of `entities` isn't included in your generator here?

Reply:
These are the contents of a message that has both a normal links and a text link:
<pre>'text': 'google.com and bing.com and link'
'entities': [
    {'type': 'url', 'offset': 0, 'length': 10},
    {'type': 'url', 'offset': 15, 'length': 8},
    {'type': 'text_link', 'offset': 28, 'length': 4, 'url': 'http://yahoo.com/'}
]</pre>

Entities for a normal URL's don't have a "url" field, so I can't just get the 'url' field of all entities. Meanwhile, text links, when parsed with `parse_entities`, will only show the text from the message, not the actual link (in this example it would parse to "link" instead of "yahoo.com"), so I can't use the parse method to get text links either. I don't see how the URLs can all be extracted in a single pass...